internal/Get-XpDirTreeRestoreFile.ps1

```diff
@@ -63,7 +63,7 @@ function Get-XpDirTreeRestoreFile {
 	ForEach ($d in $dirs) {
 		$fullpath = "$path$($d.Subdirectory)"
 		Write-Message -Level Verbose -Message "Enumerating subdirectory '$fullpath'"
-		$Results += Get-XpDirTreeRestoreFile -path $fullpath -SqlInstance $SqlInstance
+		$Results += Get-XpDirTreeRestoreFile -path $fullpath -SqlInstance $Server -SqlCredential $SqlCredential
```

Comment:
Does this need the credential as we're passing through a server object?

Reply:
technically no but why not ?

Reply:
More me checking I'd done the right thing when I took the credential passes out when I streamlined the connection stuff (your comment reminded me), and wanted to make sure I wasn't going to cause problems later one.

Reply:
It should actually ignore the credential in this case. The Connect-SqlInstance command validates whether an SMO object was passed in and if it was it goes with that...have not specifically debugged it to watch but recall it does not acknowledge the credential.

Reply:
It would be standard among other commands that we do not re-pass in that credential object if you are using the server object. So would be cleaner to remove it so it is understood that it is not in use.